### PR TITLE
Enhancement/add unit tests

### LINKED ISSUE #32: [FEATURE] Write tests

### PR DESCRIPTION
**Pull request by fairyhunter13**
**IMPORTANT: Please do not create a Pull Request without creating an issue first!**

_(Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request)._

### Summary

_Explains the information and/ motivation for making this changes..._

Adding partial unit tests on the `matchers` package

### Proposed of changes

This PR fixes/implements the following **bugs/features**:

- Write tests on issue #32 

### How has this been tested?

Proof:

```
Running tool: C:\Go\bin\go.exe test -timeout 30s -coverprofile=C:\msys64\tmp\vscode-goKE6Jxj\go-code-cover ktbs.dev/teler/pkg/matchers -race -v

=== RUN   TestIsLogformat
=== RUN   TestIsLogformat/invalid_log_format
[[31mERR[0m] Error! Only validates log format; please check your config file
[[34mINF[0m] Use "-h" flag for more info about command.
=== RUN   TestIsLogformat/no_$_sign
[[31mERR[0m] Error! Only validates log format; please check your config file
[[34mINF[0m] Use "-h" flag for more info about command.
=== RUN   TestIsLogformat/nonexist_required_log_format
[[31mERR[0m] Error! Only validates log format; please check your config file
[[34mINF[0m] Use "-h" flag for more info about command.
=== RUN   TestIsLogformat/valid_log_format
--- PASS: TestIsLogformat (0.00s)
    --- PASS: TestIsLogformat/invalid_log_format (0.00s)
    --- PASS: TestIsLogformat/no_$_sign (0.00s)
    --- PASS: TestIsLogformat/nonexist_required_log_format (0.00s)
    --- PASS: TestIsLogformat/valid_log_format (0.00s)
=== RUN   TestIsToken
=== RUN   TestIsToken/invalid_token
[[31mERR[0m] Error! Only validates token; please check your config file
[[34mINF[0m] Use "-h" flag for more info about command.
=== RUN   TestIsToken/valid_slack_token
=== RUN   TestIsToken/valid_telegram_token
--- PASS: TestIsToken (0.00s)
    --- PASS: TestIsToken/invalid_token (0.00s)
    --- PASS: TestIsToken/valid_slack_token (0.00s)
    --- PASS: TestIsToken/valid_telegram_token (0.00s)
=== RUN   TestIsHexcolor
=== RUN   TestIsHexcolor/invalid_hex_color
[[31mERR[0m] Error! Only validates hex color; please check your config file
[[34mINF[0m] Use "-h" flag for more info about command.
=== RUN   TestIsHexcolor/invalid_hex_syntax
[[31mERR[0m] Error! Only validates hex color; please check your config file
[[34mINF[0m] Use "-h" flag for more info about command.
=== RUN   TestIsHexcolor/valid_hex_color
--- PASS: TestIsHexcolor (0.00s)
    --- PASS: TestIsHexcolor/invalid_hex_color (0.00s)
    --- PASS: TestIsHexcolor/invalid_hex_syntax (0.00s)
    --- PASS: TestIsHexcolor/valid_hex_color (0.00s)
=== RUN   TestIsParseMode
=== RUN   TestIsParseMode/invalid_parse_mode
[[31mERR[0m] Error! Only validates parse mode [Markdown(v2), or HTML]; please check your config file
[[34mINF[0m] Use "-h" flag for more info about command.
=== RUN   TestIsParseMode/outdated_markdown
[[31mERR[0m] Error! Only validates parse mode [Markdown(v2), or HTML]; please check your config file
[[34mINF[0m] Use "-h" flag for more info about command.
=== RUN   TestIsParseMode/case_sensitive_invalid_html
[[31mERR[0m] Error! Only validates parse mode [Markdown(v2), or HTML]; please check your config file
[[34mINF[0m] Use "-h" flag for more info about command.
=== RUN   TestIsParseMode/valid_markdown
=== RUN   TestIsParseMode/valid_html
--- PASS: TestIsParseMode (0.00s)
    --- PASS: TestIsParseMode/invalid_parse_mode (0.00s)
    --- PASS: TestIsParseMode/outdated_markdown (0.00s)
    --- PASS: TestIsParseMode/case_sensitive_invalid_html (0.00s)
    --- PASS: TestIsParseMode/valid_markdown (0.00s)
    --- PASS: TestIsParseMode/valid_html (0.00s)
=== RUN   TestIsChannel
=== RUN   TestIsChannel/invalid_channel_format
[[31mERR[0m] Error! Only validates Slack channel; please check your config file
[[34mINF[0m] Use "-h" flag for more info about command.
=== RUN   TestIsChannel/invalid_lowercase_channel_name
[[31mERR[0m] Error! Only validates Slack channel; please check your config file
[[34mINF[0m] Use "-h" flag for more info about command.
=== RUN   TestIsChannel/valid_channel_format
--- PASS: TestIsChannel (0.00s)
    --- PASS: TestIsChannel/invalid_channel_format (0.00s)
    --- PASS: TestIsChannel/invalid_lowercase_channel_name (0.00s)
    --- PASS: TestIsChannel/valid_channel_format (0.00s)
=== RUN   TestIsChatID
=== RUN   TestIsChatID/invalid_chat_id
[[31mERR[0m] Error! Only validates chat_id; please check your config file
[[34mINF[0m] Use "-h" flag for more info about command.
=== RUN   TestIsChatID/digit_chat_id
--- PASS: TestIsChatID (0.00s)
    --- PASS: TestIsChatID/invalid_chat_id (0.00s)
    --- PASS: TestIsChatID/digit_chat_id (0.00s)
PASS
coverage: 100.0% of statements
ok  	ktbs.dev/teler/pkg/matchers	1.052s	coverage: 100.0% of statements
```

### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [x] I have followed the guidelines in our [CONTRIBUTING.md](https://github.com/kitabisa/teler/blob/development/.github/CONTRIBUTING.md) document.
- [x] I have written new tests for my changes.
  - [x] My changes successfully ran and pass tests locally.

### CODE REVIEW

**Comment by fairyhunter13** [comment]
@dwisiswant0 Could you please check my code?

**Review by dwisiswant0 — CHANGES_REQUESTED** [review]
Hi @fairyhunter13, I confirm to request changes & some revisions for tests you are working on.

Please note that the closing issue that you mention is for the whole part of the test, considering this PR is only partially done, I suggest to change this summary **TO NOT** use closing keywords - but you can still link to issue #32 to track the progress of that issue ticket.

File: pkg/matchers/config_test.go [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+		},
+		{
+			name: "no $ sign",
+			args: args{
+				s: "remote_addr",
+			},
+			wantErr: true,
+		},
+		{
+			name: "nonexist required log format",
```
[the remark this answers is not in the document]
Comment (fairyhunter13) [reply pkg/matchers/config_test.go:33]: Ok, I will remove this part.

File: pkg/matchers/config_test.go [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+		s string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "invalid log format",
+			args: args{
```
[the remark this answers is not in the document]
Comment (fairyhunter13) [reply pkg/matchers/config_test.go:20]: @dwisiswant0 Do I need to add trailing space after `args`? I think the go standard linting automatically remove space after `args` and before `{`. Or maybe do you have any linting settings available especially for this project?

[the remark this answers is not in the document]
Comment (dwisiswant0) [reply pkg/matchers/config_test.go:20]: Nope. It's just a code-style matter of this project. I untick that first checklist in this PR summary, so you can tick again if you feel you have complied.

[the remark this answers is not in the document]
Comment (fairyhunter13) [reply pkg/matchers/config_test.go:20]: @dwisiswant0 Sorry, if I miss something, where do I can find the code-style matter of this project?

[the remark this answers is not in the document]
Comment (dwisiswant0) [reply pkg/matchers/config_test.go]: Currently is set in stone and _stick a fork in it_ living documents later, and will be added at [Our Responsibilities](https://github.com/kitabisa/teler/blob/master/.github/CODE_OF_CONDUCT.md#our-responsibilities) part of [Code of Conduct](https://github.com/kitabisa/teler/blob/master/.github/CODE_OF_CONDUCT.md).

File: pkg/matchers/config_test.go [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+			name: "invalid lowercase channel name",
+			args: args{
+				s: "kitabisa13",
+			},
+			wantErr: true,
+		},
+		{
+			name: "valid channel format",
+			args: args{
+				s: "KITABISA13",
```
[the remark this answers is not in the document]
Comment (dwisiswant0) [reply pkg/matchers/config_test.go:229]: Edited.

**Comment by fairyhunter13** [comment]
@dwisiswant0 Could you please check my PR again?

**Review by dwisiswant0 — APPROVED** [review]

**Comment by dwisiswant0** [comment]
This looks lit @fairyhunter13. Mad thanks for your contributions & of course for your efforts, merging soon. :fire: 
@all-contributors please add @fairyhunter13 for test.

**Comment by fairyhunter13** [comment]
@dwisiswant0 Sorry for my inexperience in contributing to the open-source project. Thank you for your reviews and hard work. Please keep it up! 👍 👍 👍

**Pull request merged**